### PR TITLE
Fix largest logit selection in sync.gd

### PR DESCRIPTION
**Pull request by Ivan-267**
Should select the largest logit correctly when logits are below 0.

Not tested yet.

### CODE REVIEW

**Comment by Ivan-267** [comment]
**Update:**
Added support for sampling actions from the stochastic policy when doing onnx inference (instead of selecting the largest logit, will select an action based on the probabilities). 

image

**Reasoning:**
In some cases, in partially observable environments, stochastic action selection may prevent the agent from getting stuck (while usually it will not appear as smooth and optimal performance and safely avoiding obstacles is not guaranteed). It should more closely match the trained policy performance observed during training, especially if the trained policy still has a relatively high entropy.

**Limitations**:
In current implementation, I just added this to discrete actions where we deal with logits manually. This applies to SB3 PPO with discrete actions, and would apply to other scripts that export logits from a stochastic policy (e.g. if we added a CleanRL discrete actions PPO variant), I think it also applies to Rllib but have not verified yet. Sample Factory's onnx export works differently, but we haven't fully implemented support for it yet (might need adjustments on either their export script or the C# inference code).

For continuous actions, we have different outputs from SB3/Rllib example script exported onnx files, so stochastic sampling is not yet added (SB3 onnx export script could be adapted to export a format more similar to Rllib, but that's just one possible approach not without any downsides). Adding it to the gdscript inference code right now would enable it only for Rllib's export, which can be considered for later updates.

**Tests**: 
This addition is not yet tested, will update the comment when I have some time. I've added it to this PR so that both changes can be reviewed together before merging.

**Comment by Ivan-267** [comment]
Update: Renaming the branch (as I'm using it commonly in new envs and env updates) to make it more clear what it does closed the PR, so I kept the original name.
